### ISSUE TITLE
 ARR Showcase App - Failed to load model

### ISSUE THREAD

**Issue by selinkampa** [Azure/azure-remote-rendering#44]
Hi,

* In Remote Rendering Showcase app, models from our blob container are getting listed under the menu but aren't loading when user press the load buttons. It shows error:  Failed to load model. Reason: Load model failed with status 'FileDownloadFailed' 

Model names are listed correctly, and session is connecting without issues.
The issue started happening since 2-3 weeks. Before we were able to load models from blob container without issues.
Models are converted using ARRT, there in ARRT it is loading and rendering fine.

As I was using the release of Showcase app from some months ago, I thought maybe building it again might help, I pulled the latest release and took build, but no luck and still the same error.

* The sessionId:Arr62aa7224c6db19a77341a90b69b05440

I also have given storage account access to remote rendering account, following this documentation https://docs.microsoft.com/en-us/azure/remote-rendering/how-tos/create-an-account#link-storage-accounts:
![RR](https://user-images.githubusercontent.com/89584542/131326764-13020b65-62e4-40b3-9b55-dce337f3f42b.png)
![rr2](https://user-images.githubusercontent.com/89584542/131326393-0c763078-a7f3-4690-8983-88f433594114.png)

Thanks in advance.

**Comment by ChristopherManthei** [comment]
Hi @selinkampa,

thank you for the report. The reason ARRT works and Showcase not is most likely because ARRT uses LoadModelFromSasAsync and Showcase uses LoadModelAsync. So there is probably a problem with the storage account linkage.
We are looking into the issue right now and will get back to you as soon as we can.

Cheers,
Christopher

**Comment by ChristopherManthei** [comment]
It looks like the ARR account was moved to another subscription. In that case, the problem can probably be fixed by turning the system assigned managed identity off and on again:
![troubleshoot-portal-identity](https://user-images.githubusercontent.com/22638947/131369866-4262b0fc-9f1f-4266-a291-3814173f3d3c.png)
You might need to [re-link your storage account](https://docs.microsoft.com/en-us/azure/remote-rendering/how-tos/create-an-account#link-storage-accounts) with your arr account after doing this.

**Comment by jeofyoung** [comment]
Just wanted to add some info in addition to what @ChristopherManthei already covered:

- From discussions in a separate thread, my understanding is your Azure resources were moved to a different org / tenant ? This would explain why the service is now unable to access storage. Unfortunately, attached identities cannot move with the resource as mentioned here: [List role assignments for managed identities](https://docs.microsoft.com/en-us/azure/role-based-access-control/transfer-subscription#list-role-assignments-for-managed-identities)

> Managed identities do not get updated when a subscription is transferred to another directory. As a result, any existing system-assigned or user-assigned managed identities will be broken. After the transfer, you can re-enable any system-assigned managed identities.

  The same note applies even if only resources were moved to a subscription in a different tenant instead of the whole subscription.

- Also, managed identities cannot be used to access resources in a different tenant as mentioned here: [Can I use a managed identity to access a resource in a different directory/tenant?](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/managed-identities-faq#can-i-use-a-managed-identity-to-access-a-resource-in-a-different-directorytenant)

> Managed identities do not currently support cross-directory scenarios.


**To summarize:**

- Both your ARR account/ resource and your storage account needs to be in the same tenant
- Do toggle "System assigned" which will create and attach a new identity (with the same name but different ID in the new tenant) then you'll need to re-assign permissions to your storage account as per: [re-link your storage account](https://docs.microsoft.com/en-us/azure/remote-rendering/how-tos/create-an-account#link-storage-accounts).

Apologies for the inconvenience.

**Comment by selinkampa** [comment]
Dear @ChristopherManthei and @jeofyoung,
thank you so much for the support.

It was indeed about the migration of the tenant, there must be something going wrong during that. 

First I tried toggling "System assigned", which did remove the "undefined identity" and the name of the remote rendering account was shown in the assigned roles list again. But that didn't fix the issue, and I still couldn't load models.

In the separate thread, I received one more comment additional to what you said, which is to open a new remote rendering account if toggling wouldn't help. Once I opened a second remote rendering account in the new tenant and linked it with the existent blob storage, and took another Unity build with the new keys, it worked! We can render remotely again :)

Great work and support! Thanks once more.

**Comment by jeofyoung** [comment]
Hi @selinkampa,

Thank you very much for reporting your results after re-creating the identity. It seems that something we enabled recently is causing an issue that prevented the newly created identity for your old account from getting registered which we will resolve. Thanks for helping us uncover this!

**Comment by jeofyoung** [comment]
The issue preventing a re-created identity from being registered in the system has been resolved.
